src/main/java/com/microsoft/sqlserver/jdbc/SharedTimer.java

```diff
@@ -106,14 +110,20 @@ public void removeRef() {
      * When the caller is finished with the SharedTimer it must be released via {@link#removeRef}
      */
     public static SharedTimer getTimer() {
-        synchronized (lock) {
-            if (instance == null) {
-                // No shared object exists so create a new one
-                instance = new SharedTimer();
+        SharedTimer result = instance;
```

Comment:
Similar comment to above, why can't this just be wrapped in a try / catch like the other changes?

Reply:
Refer to https://en.wikipedia.org/wiki/Double-checked_locking

"Note the local variable "localRef", which seems unnecessary. The effect of this is that in cases where helper is already initialized (i.e., most of the time), the volatile field is only accessed once (due to "return localRef;" instead of "return helper;"), which can improve the method's overall performance by as much as 40 percent."

One could use the VarHandles stuff for even better performance but that's even trickier.